test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -105,6 +105,45 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Tests
 
+(deftest validate-settings-test
+  (testing "invalid ca-ttl is rejected"
+    (let [settings (assoc
+                     (testutils/ca-settings cadir)
+                     :ca-ttl
+                     (+ max-ca-ttl 1))]
+      (is (thrown? IllegalStateException (validate-settings! settings)))))
```

Comment:
Since `IllegalStateExecption` is so generic, I think it's preferable here to use `thrown-with-msg?` and test for the specific error message we expect.  I like to avoid that sort of thing when possible, but without it here, there's no guarantee that this `IllegalStateException` is the one we're expecting.